hugr-core/src/hugr/views/sibling_subgraph.rs

```diff
@@ -340,13 +340,22 @@ impl SiblingSubgraph {
         let rep_root = replacement.root();
         let dfg_optype = replacement.get_optype(rep_root);
         if !OpTag::Dfg.is_superset(dfg_optype.tag()) {
-            return Err(InvalidReplacement::InvalidDataflowGraph);
+            return Err(InvalidReplacement::InvalidDataflowGraph {
+                node: rep_root,
+                op: dfg_optype.clone(),
+            });
         }
         let Some([rep_input, rep_output]) = replacement.get_io(rep_root) else {
-            return Err(InvalidReplacement::InvalidDataflowParent);
+            return Err(InvalidReplacement::InvalidDataflowParent {
```

Comment:
This feels a strangely named error. It's not that the optype isn't a `DataflowParent` - that's guaranteed from the previous check that it's a Dfg. (Odd that we are checking `OpTag::Dfg.is_superset(dfg_optype)` rather than `OpTag::Dfg == dfg_optype` but there.)

So, it's that the input/output are missing. Indeed that's what the `#[error]` message for `InvalidDataflowParent` says! I'd suggest renaming to, say, `MissingInputOutput`.

(Also we kind of know what the `op` is, it's an `OpType::DFG` - so this just puts the input/output *typerows* in, well I guess that's fair.)

Reply:
I think we can replace this error with an `expect`. It can only trigger with invalid hugrs.